crates/weaver_live_check/src/lib.rs

```diff
@@ -584,6 +599,9 @@ impl LiveCheckStatistics {
 
     /// Increment the total number of entities by type
     pub fn inc_entity_count(&mut self, entity_type: &str) {
+        if self.disabled {
```

Comment:
Would it be easier to make this a `dyn` trait and have a "disable" instance and then the real thing and just return the empty instance?

Example:

```rust
pub trait LiveCheckStatistics {
   fn inc_entity_count(...);
   ...
}

struct NoStatistics {}
impl LiveCheckStatistics for NoStatistics { ... }

struct CumualtiveStatistics {}
impl LiveCheckStatistics for CumulativeStatistics { ... }

fn my_live_check_runner() {
  let stats: Box<dyn LiveCheckStatistics> = Box::new(...);
}
```